### ISSUE TITLE
Global Methods dont work with out type

### ISSUE THREAD

**Issue by becjit** [netglade/auto_mappr#168]
**Describe the bug**
I have multiple mappers that uses dame custom field mapping. Lets say there are multiple fields across mappers where I use same custom converters.
In this case static methods without any Source model argument is required, but looks like converters are not allowed without the source model
For example :
From your example I cant make 
 `int mapAge(UserDto _) => 42;`

as 
 `int mapAge() => 42;`

**To Reproduce**
Code to reproduce issue: 

**Expected behavior**
Global methods should support methods without any source model

**Version info**
 - Package version : 

**Additional context**

**Comment by tenhobi** [comment]
Hi @becjit, could you please elaborate more?

> In this case static methods without any Source model argument is required, but looks like converters are not allowed without the source model

So you would want a type converter that does not have a source? How would that work?

---

The correct/current behavior is like this and I think it makes sense, no? For you to map/convert, you always need the source to produce the target. How would that work otherwise -- matching which converter to use, for example? 

```dart
@AutoMappr(
  [
    MapType<UserDto, User>(
      // MapType specific
      converters: [
        TypeConverter<String, DateTime>()
      ],
    ),
  ],
  converters: [
    // or globally here
  ],
)
class Mappr extends $Mappr {
  static DateTime stringToDateTime(String source) {
    // ...
  }
}
```

**Comment by becjit** [comment]
Sure thing! 
Let's say in the target I have a field with DateTime, and source does not have any corresponding column.
So I really dont need the source model to generate value for the target; a simple DateTime.now() would do.

Now let's say I have multiple models with similar requirements. Currently I am copy pasting the same code with separate SourceModel as type converter across mappers

For example


```
@AutoMappr(
  [
    MapType<UserDto, User>(
      // MapType specific
      converters: [
        TypeConverter<String, DateTime>()
      ],
    ),
  ],
  converters: [
    // or globally here
  ],
)
class Mappr extends $Mappr {
  static DateTime stringToDateTime(/* not needed */String source) {
    return DateTime.now();
  }
}

```

**Comment by tenhobi** [comment]
I see. Well, since `TypeConverter` converts from source and target, that's not the right thing to do anyway.

We could however potentially extend the behavior of `custom` in `Field`, since even now (I think :D) you can use either a function `TARGET something(SOURCE)` or a `TARGET` constant. And I think we could distinguish between 0 and 1 argument there based on the function you pass in. 🤔 

Aka then you could do this

```dart
@AutoMappr([
  MapType<OtherDto, Other>(
    fields: [
      Field('something', custom: Utility.mapStringToInt), // internally will use Utility.mapStringToInt(model)
      Field('created', custom: Utility.mapDateTime), // internally will use Utility.mapDateTime()
    ],
  ),
])
class OtherMappr extends $Mappr {
  const OtherMappr();
}

class Utility {
  static DateTime mapDateTime() {
    return DateTime.now();
  }
  static String mapIntToString(String value) {
    return '$value';
  }
}
```

Thoughts, @petrnymsa ?

---

That being said, this is potentially an easy upgrade, do you want to take this and do a PR for that? Otherwise we don't probably have time any time soon for this -- maybe in January or February.

What is needed to be done:
- https://github.com/netglade/auto_mappr/blob/main/packages/auto_mappr/lib/src/extensions/dart_object_extension.dart#L19C52-L25 modify this (detect 0 vs 1 arguments)
- add some tests

**Comment by petrnymsa** [comment]
Agree, that we can extend "custom" behavior on Field level in way that

- If provided function has any argument - it has to be "SOURCE" object argument - e.g for custom mapping from source (**current behavior**
- If provided function has zero arguments - call it as is. **new behavior**. This would allow OP's request.

**Comment by petrnymsa** [comment]
I am cooking some ad-hoc solution right now 🔥

**Comment by petrnymsa** [comment]
@becjit Can you try out this solution from #169 ?

In pubspec.yaml you can change auto_mappr dependency from 

```yaml

auto_mappr: ^x.xx
```
to
```
auto_mappr:
    git:
      url: https://github.com/netglade/auto_mappr
      path: packages/auto_mappr
      ref: feat/168-extends-custom-mapping
```

Let us know! Thanks 🚀

**Comment by becjit** [comment]
Sure thing. Thanks a lot. Will get back to you soon

**Comment by tenhobi** [comment]
Hey @becjit, did you have time to validate whether this change helps your case? :)

**Comment by petrnymsa** [comment]
Fix available in 2.3.0 🎉
